frontend/src/components/dashboard/home/VisitsDensityWidget.vue

```diff
@@ -0,0 +1,205 @@
+        </div>
+        <VueApexCharts
+            type="heatmap"
+            height="350"
+            :options="chartOptions"
+            :series="drawHeatmap"
+            class="visits-heatmap"
+        />
+        <VisitsDensityPeriodDropdown
+            class="ml10"
+            :value="getSelectedPeriod"
+            @change="changeSelectedPeriod"
+        />
+    </div>
+</template>
+
+<script>
+    import VueApexCharts from 'vue-apexcharts';
+    import VisitsDensityPeriodDropdown from "@/components/dashboard/common/VisitsDensityPeriodDropdown";
+    import {mapGetters, mapActions} from 'vuex';
+    import {
+        GET_SELECTED_PERIOD,
+        GET_VISITS_DATA,
+        IS_FETCHING
+    } from "@/store/modules/visits_density_widget/types/getters";
+    import {
+        CHANGE_SELECTED_PERIOD,
+        FETCH_WIDGET_DATA
+    } from "@/store/modules/visits_density_widget/types/actions";
+    import Spinner from "@/components/utilites/Spinner";
+
+    export default {
+        name: "VisitsDensityWidget",
+        components: {
+            VueApexCharts,
+            VisitsDensityPeriodDropdown,
+            Spinner
+        },
+        data() {
+            return {
+                days: {
+                    '0': 'Sun',
+                    '1': 'Mon',
+                    '2': 'Tue',
+                    '3': 'Wed',
+                    '4': 'Thu',
+                    '5': 'Fri',
+                    '6': 'Sat'
+                },
+                hours: {
+                    '0': '12am',
+                    '2': '2am',
+                    '4': '4am',
+                    '6': '6am',
+                    '8': '8am',
+                    '10': '10am',
+                    '12': '12pm',
+                    '14': '2pm',
+                    '16': '4pm',
+                    '18': '6pm',
+                    '20': '8pm',
+                    '22': '10pm'
+                },
+                chartOptions: {
+                    chart: {
+                        toolbar: {
+                            show: false
+                        },
+                        fontFamily: 'Gilroy'
+                    },
+                    colors: ["#3C57DE"],
+                    dataLabels: {
+                        enabled: false
+                    },
+                    grid: {
+                        yaxis: {
+                            lines: {
+                                show: false
+                            }
+                        },
+                        padding: {
+                            left: 15
+                        },
+                    },
+                    legend: {
+                        show: false
+                    },
+                    plotOptions: {
+                        heatmap: {
+                            radius: 6,
+                        }
+                    },
+                    states: {
+                        hover: {
+                            filter: {
```

Comment:
it is not your changes, please check in develop is it correct

Reply:
the same is in develop